### ISSUE TITLE
RSyntaxTextArea still displays caret when `setFocusable(false)`

### ISSUE THREAD

**Issue by DJ-Raven** [bobbylight/RSyntaxTextArea#729]
**Description**
When an RSyntaxTextArea is configured with setFocusable(false), it cannot receive keyboard focus, but its caret is still visible.
This makes the text area appear as if it is editable/focused even though it is not focusable.

**Steps to Reproduce**
You can test this code
```java
import org.fife.ui.rsyntaxtextarea.RSyntaxTextArea;
import org.fife.ui.rtextarea.RTextScrollPane;

import javax.swing.*;
import java.awt.*;

public class Test extends JFrame {

	public Test() {
		setDefaultCloseOperation(EXIT_ON_CLOSE);
		setSize(400, 400);
		setLocationRelativeTo(null);

		RSyntaxTextArea textArea1 = new RSyntaxTextArea();
		RSyntaxTextArea textArea2 = new RSyntaxTextArea();

		RTextScrollPane scrollPane1 = new RTextScrollPane(textArea1);
		RTextScrollPane scrollPane2 = new RTextScrollPane(textArea2);
		add(new JTextField(), BorderLayout.NORTH);

		JPanel panel = new JPanel(new GridLayout(1, 2));

		textArea1.setFocusable(false);
		textArea2.setFocusable(false);
		panel.add(scrollPane1);
		panel.add(scrollPane2);
		add(panel, BorderLayout.CENTER);
	}

	public static void main(String[] args) {
		SwingUtilities.invokeLater(() -> new Test().setVisible(true));
	}
}
```
**Expected behavior**
When an RSyntaxTextArea is not focusable, I would expect its caret to not be displayed.

**Actual behavior**
The caret is still visible in the RSyntaxTextArea components even though they are not focusable.

**Screenshots**

<img width="416" height="428" alt="Image" src="https://github.com/user-attachments/assets/0685204e-582e-4090-ac86-154bc79a611a" />

**Java version**
Tested on JetBrains Runtime 25 and other JDK distributions. The same behavior occurs on all tested JDKs.

Thank you!
